src/renderers/shared/fiber/ReactFiberStack.js

```diff
@@ -0,0 +1,102 @@
+/**
+ * Copyright 2013-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @providesModule ReactFiberStack
+ * @flow
+ */
+
+'use strict';
+
+import type { Fiber } from 'ReactFiber';
+
+export type StackCursor = {|
```

Comment:
We should make this generic `type StackCursor<T> = {| current: T |}`.

You will have to make some unsound things when you pop here but then everywhere else is fine.